testing/tests/DevExpress.ui.widgets/treeViewParts/searching.js

```diff
@@ -497,3 +497,38 @@ QUnit.test('apply search after searchTimeout', function(assert) {
     assert.equal($items.length, 4, 'filter was applied after timeout');
     this.clock.restore();
 });
+
+['none', 'selectAll', 'normal'].forEach(selectionMode => {
+    QUnit.test(`focusIn -> search -> focusIn -> clearSearch -> focusIn. selectionMode: ${selectionMode}`, function(assert) {
+        const treeView = initTree({
+            items: [{ id: 1, text: 'item1' }, { id: 2, text: 'item2' }],
+            searchEnabled: true
+        }).dxTreeView('instance');
+
+        const isNodeFocused = (id) => treeView.$element().find(`[data-item-id="${id}"]`).hasClass('dx-state-focused');
+        const triggerFocus = () => treeView._focusInHandler({});
```

Comment:
_focusInHandler is not used in other tests

$(this.$element).trigger('focusin'); ?
$('#input1').focus(); ?
QUnit.testInActiveWindow ?
$parentItem.trigger('dxpointerdown'); ?
wrapper.instance._moveFocus(config.direction, {}); ?
$treeView.focusin(); ?
https://js.devexpress.com/Documentation/ApiReference/UI_Components/dxTreeView/Methods/#focus ?
First list item should be focused on the \'tab\' key press when the search editor is focused ?

Reply:
> https://js.devexpress.com/Documentation/ApiReference/UI_Components/dxTreeView/Methods/#focus 
Current implementation of this method works as follows: if widget has text field, then textEditor must be  focusedhttps://github.com/DevExpress/DevExtreme/blob/21_2/js/ui/widget/ui.search_box_mixin.js#L157

When customer press 'tab', focus moves to next active target. In next commit I simulate it via eventsEngine as follows https://github.com/DevExpress/DevExtreme/pull/19028/commits/a39f7624a14fc0627e025e63dcceab4bcb556348#diff-3f5aa2a335b4f895ae3794b42d3898ac3145fee4b03edeaaa70023c45f088b3dR513